src/extensions/struct-conn/index.ts

```diff
@@ -0,0 +1,365 @@
+export const StructConnExtensionFunctions = {
+    /** Return an object with all struct_conn records for a loaded structure.
+     * Applies to the first structure belonging to `entry` (e.g. '1tqn'),
+     * or to the first loaded structure overall if `entry` is `undefined`.
+     */
+    getStructConns(plugin: PluginContext, entry: string | undefined): { [id: string]: StructConnRecord } {
+        const structNode = selectStructureNode(plugin, entry);
+        const structure = structNode?.obj?.data;
+        if (structure) return extractStructConns(structure);
+        else return {};
+    },
+
+    /** Create visuals for residues and atoms involved in a struct_conn with ID `structConnId`
+     * and zoom on them. If `keepExisting` is false (default), remove any such visuals created by previous calls to this function.
+     * Also hide all carbohydrate SNFG visuals (as they would occlude our residues of interest).
+     */
+    async inspectStructConn(plugin: PluginContext, entry: string | undefined, structConnId: string, keepExisting: boolean = false) {
+        const structNode = selectStructureNode(plugin, entry);
+        const structure = structNode?.obj?.data;
+        if (!structure) {
+            console.error('Structure not found');
+            return;
+        }
+
+        const structNodeRef = structNode.transform.ref;
+        const structConnCache = getExtensionState(plugin).structConnCache;
+        if (!structConnCache.has(structNodeRef)) {
+            structConnCache.set(structNodeRef, extractStructConns(structure));
+        }
+        const conns = structConnCache.get(structNodeRef)!;
+        const conn = conns[structConnId];
+        if (!conn) {
+            console.error(`The structure does not contain struct_conn "${structConnId}"`);
+            return;
+        }
+
+        if (!keepExisting) {
+            await removeAllStructConnInspections(plugin);
+        }
+        await addStructConnInspection(plugin, structNode, conn);
+        hideSnfgNodes(plugin);
+    },
+
+    /** Remove anything created by `inspectStructConn` and
+     * make visible any carbohydrate SNFG visuals that have been hidden by `inspectStructConn`.
+     */
+    async clearStructConnInspections(plugin: PluginContext) {
+        await removeAllStructConnInspections(plugin);
+        unhideSnfgNodes(plugin);
+    },
+};
+
+
+type StructNode = ReturnType<typeof selectStructureNode>
+
+/** Return the first structure node belonging to `entry` (e.g. '1tqn'),
+ * or to the first loaded structure node overall if `entry` is `undefined`.
+ * Includes only "root" structures, not structure components. */
+function selectStructureNode(plugin: PluginContext, entry: string | undefined) {
+    const structNodes = plugin.state.data
+        .selectQ(q => q.ofType(PluginStateObject.Molecule.Structure))
+        .filter(node => node.obj && !node.obj.data.parent && !node.transform.transformer.definition.isDecorator);
```

Comment:
```suggestion
        .selectQ(q => q.ofType(PluginStateObject.Molecule.Structure))
        .filter(node => node.obj && !node.obj.data.parent && !node.transform.transformer.definition.isDecorator);
```

You should filter on `PluginStateObject.Molecule.Model` directly as that is where you are looking for the id anyway. Then you can find a suitable child structure node to create your representation on. `plugin.managers.structure.hierarchy` should help there quite a bit.

Reply:
I don't see how it would make things easier. It would add extra work of finding the suitable child structure node.

Reply:
And now you are doing an extra work to incorrectly filter out the decorators. You should actually be applying your selection to the lowest decorator of the root, and not the root itself. There could be things like transforms in the decorators.

Reply:
I checked it now, and even though I apply the selections to the root structure, in reality they get applied to the decorator. Is there maybe some smart magic going on?

You can see, that the selections are hanging on `0LcJJj-WDrnJZbPNBuKi-A`, not `eYTXqUzrVd9dEiAWUgQA0g`:
```
              {
                "parent": "tA8TVGNV9FUPCqdX-KZY4Q",
                "transformer": "ms-plugin.structure-from-model",
                "ref": "eYTXqUzrVd9dEiAWUgQA0g",
              },
              {
                "parent": "eYTXqUzrVd9dEiAWUgQA0g",
                "transformer": "ms-plugin.custom-structure-properties",
                "ref": "0LcJJj-WDrnJZbPNBuKi-A",
              },
              {
                "parent": "0LcJJj-WDrnJZbPNBuKi-A",
                "transformer": "ms-plugin.structure-component",
                "tags": [
                  "structconn-focus-residue-sel"
                ],
                "ref": "AWuKbfKGb0LAkHLoEAN_ZQ",
              },
              {...},
              {
                "parent": "0LcJJj-WDrnJZbPNBuKi-A",
                "transformer": "ms-plugin.structure-component",
                "tags": [
                  "structconn-focus-atom-sel"
                ],
                "ref": "YdB3_WI7VXbrfRsd2GaAeQ",
              },
              {...}
```